community/lld/APKBUILD

```diff
@@ -1,42 +1,45 @@
 # Contributor: Jakub Jirutka <jakub@jirutka.cz>
 # Maintainer: Jakub Jirutka <jakub@jirutka.cz>
 pkgname=lld
-pkgver=4.0.0
+pkgver=5.0.0
 pkgrel=0
-_vermajor=${pkgver%%.*}
+_llvmver=${pkgver%%.*}
 pkgdesc="The LLVM Linker"
 url="http://llvm.org"
 arch="all"
 license="UOI-NCSA"
 makedepends="
 	cmake
-	llvm-dev>=$_vermajor
-	llvm-static>=$_vermajor
+	libedit-dev
+	llvm-dev>=$_llvmver
+	llvm-static>=$_llvmver
+	llvm-test-utils=>$_llvmver
 	zlib-dev"
+checkdepends="gtest gtest-dev"
 subpackages="$pkgname-dev"
-source="http://llvm.org/releases/$pkgver/$pkgname-$pkgver.src.tar.xz
-	cmake-fix-pthread-handling-for-out-of-tree-builds.patch"
+source="https://llvm.org/releases/$pkgver/$pkgname-$pkgver.src.tar.xz"
 builddir="$srcdir/$pkgname-$pkgver.src"
 
 build() {
 	mkdir -p "$builddir"/build
 	cd "$builddir"/build
 
 	cmake .. \
-		-DCMAKE_BUILD_TYPE=Release \
-		-DCMAKE_VERBOSE_MAKEFILE=OFF \
+		-DCMAKE_BUILD_TYPE=MinSizeRel \
+		-DCMAKE_C_FLAGS_MINSIZEREL_INIT="$CFLAGS" \
+		-DCMAKE_CXX_FLAGS_MINSIZEREL_INIT="$CXXFLAGS" \
+		-DCMAKE_EXE_LINKER_FLAGS_MINSIZEREL_INIT="$LDFLAGS" \
 		-DCMAKE_INSTALL_PREFIX=/usr \
-		-DCMAKE_CXX_FLAGS="$CXXFLAGS" \
-		-DCMAKE_C_FLAGS="$CFLAGS" \
+		\
 		-DCMAKE_SKIP_INSTALL_RPATH=ON \
-		-DLLVM_LINK_LLVM_DYLIB=ON
```

Comment:
@xentec Why did you remove this?

Reply:
I've already described it in the OP. Have you build `lld` and run it's tests without errors?

Reply:
Ah, sorry, I've overlooked it. No, still build errors from linker. Currently I'm trying to build clang with shared libs and link lldb with clang dynamically when linking with static fails.